src/frontend/src/lib/components/guard/AddressGuard.svelte

```diff
@@ -3,19 +3,21 @@
 	import { isNullish } from '@dfinity/utils';
 	import { certifyAddress } from '$lib/services/address.services';
 	import { warnSignOut } from '$lib/services/auth.services';
+	import { addressCertified } from '$lib/derived/address.derived';
+	import { ETHEREUM_TOKEN_ID } from '$env/tokens.env';
 
 	const validateAddress = async () => {
-		if (isNullish($addressStore)) {
+		if (isNullish($addressStore?.[ETHEREUM_TOKEN_ID])) {
```

Comment:
Can we use `$ethAddress` here? I'm not sure but, if possible, it would be a bit more readable.

Reply:
right! I am just in doubt about possible infinite loops: `validateAddress` will update `addressStore`, that then updates `ethAddress`. The loop should be interrupted right away, but could there be a situation with infinite loops?

Reply:
but thinking better, this was happening before too

Reply:
I don't think there was a loop. It's normal to update the story if query or update. Where was the loop?

Reply:
Not a loop, but a re-calling of the function. Below the original code:

```typescript
const validateAddress = async () => {
.................

// step 3: the recalled function exits here
if ($addressStore.certified === true) {
	// The address is certified, all good
	return;
}

// step 1: this modifies $addressStore
const { success, err } = await certifyAddress($addressStore.address);

...............
};

// step 2: this would trigger the function above again
$: $addressStore, (async () => await validateAddress())();
```

but it seems all good. So I will use `ethAddress`

Reply:
> // step 2: this would trigger the function above again

It does trigger the function again but the function prevents loop by checking for the certification

```
if ($addressStore.certified === true) {
			// The address is certified, all good
			return;
		}
```

Reply:
perfect!